### ISSUE TITLE
Scheduled websocket upgrades get the wrong IDs

### ISSUE THREAD

**Issue by erikvk** [Starcounter/Home#412]
When scheduling a `System.Action` that contains an invokation of `Request.SendUpgrade()`, using `Scheduling.RunTask()`, the resulting WebSockets do not get their proper IDs. Here's a test application:

```csharp
public class Program
{
    private const string groupName = "testgroup";
    private static readonly List<ulong> Added = new List<ulong>();

    public static void Main()
    {
        Handle.GET("/test?{?}", (string query, Request request) =>
        {
            if (!request.WebSocketUpgrade)
                return HttpStatusCode.UpgradeRequired;

            Added.Add(request.GetWebSocketId());

            if (query == "schedule")
                Scheduling.RunTask(() => request.SendUpgrade(groupName)).Wait();
            else if (query == "taskrun")
                Task.Run(() => request.SendUpgrade(groupName)).Wait();
            else request.SendUpgrade(groupName);

            return HandlerStatus.Handled;
        });

        Handle.WebSocketDisconnect(groupName, ws =>
        {
            var id = ws.ToUInt64();
            Debug.Assert(Added.Contains(id));
            Added.Remove(id);
        });
    }
}
```
Using [`wscat`](https://www.npmjs.com/package/wscat), the following requests will set up WebSockets – the second one sending the upgrade from a scheduled task.

```
wscat -c "localhost:8080/test?"
wscat -c "localhost:8080/test?schedule"
```

When disconnecting the WebSocket from the second request, the `Debug.Assert()` fails. The value of the `id` variable in the `Handle.WebSocketDisconnect` handler is then different from what `request.GetWebSocketId()` returned in the `Handle.GET` handler.

Running the following request:

```
wscat -c "localhost:8080/test?taskrun"
```

Throws a `System.NullReferenceException` in the call to `SendUpgrade()`:

```
Message: "Object reference not set to an instance of an object."
StackTrace: "at Starcounter.Internal.SocketStruct.Init(NetworkDataStream dataStream) in C:\\TeamCity\\BuildAgent\\work\\sc-17406\\Level1\\src\\Starcounter.Internal\\SchedulerResources.cs:line 190\r\n   
at Starcounter.Request.SendUpgrade(String groupName, List`1 cookies, Dictionary`2 headers, IAppsSession session) in C:\\TeamCity\\BuildAgent\\work\\sc-17406\\Level1\\src\\Starcounter.Internal\\Http\\Request.cs:line 372\r\n   
at TestProject.Program.<>c__DisplayClass2_0.<Main>b__3() in C:\\Users\\erik\\RESTar\\TestProject\\Program.cs:line 26"
```

**Comment by erikvk** [comment]
cc @alemoi probably

**Comment by erikvk** [comment]
`Version=2.3.2.930`

**Comment by alemoi** [comment]
Thanks @erikvk for the bug! I have found an issue with scheduling and websockets. I will fix that websocket is completely scheduler-agnostic in 2.4. But for 2.3 the recommendation is that you don't `SendUpgrade` from a different scheduler. Is that something we can deal on @erikvk or that is not satisfactory?

**Comment by erikvk** [comment]
Thanks @alemoi . That works fine for me!

**Comment by alemoi** [comment]
@erikvk thanks, just let me know if something else should be checked/fixed.
Forgot to mention the commit with a fix in 2.4: 419577193f0ec60ae3e5c6cd5d579b419b85784c
